osquery/tables/utility/file.cpp

```diff
@@ -1,26 +1,46 @@
 /**
- *  Copyright (c) 2014-present, Facebook, Inc.
- *  All rights reserved.
- *
- *  This source code is licensed under both the Apache 2.0 license (found in the
- *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
- *  in the COPYING file in the root directory of this source tree).
- *  You may select, at your option, one of the above-listed licenses.
- */
-
-#include <sys/stat.h>
+*  Copyright (c) 2014-present, Facebook, Inc.
+*  All rights reserved.
+*
+*  This source code is licensed under both the Apache 2.0 license (found in the
+*  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+*  in the COPYING file in the root directory of this source tree).
+*  You may select, at your option, one of the above-listed licenses.
+*/
 
+#include "Shlwapi.h"
+#include "osquery\tables\utility\file_windows_ops.h"
 #include <boost/filesystem.hpp>
-
+#include <iomanip>
+#include <iostream>
 #include <osquery/filesystem.h>
 #include <osquery/logger.h>
 #include <osquery/tables.h>
+#include <sstream>
+#include <string>
+#include <sys/stat.h>
 
 namespace fs = boost::filesystem;
 
 namespace osquery {
+
+int getUidFromSid(PSID sid);
+int getGidFromSid(PSID sid);
+time_t FileTimeToUnixTime(FILETIME& ft);
+time_t LongIntToUnixTime(LARGE_INTEGER& ft);
+
 namespace tables {
 
+#if defined(WIN32)
```

Comment:
Let's move `kDriveLetters` to our platform abstraction `./osquery/filesystem/windows/fileops.cpp`